main/src/ca/uwaterloo/flix/language/ast/ops/SimplifiedAstOps.scala

```diff
@@ -513,4 +515,291 @@ object SimplifiedAstOps {
+        val as = args map (renameBoundVars(_, vars0, labels0))
+        Expression.InvokeMethod(method, e, as, tpe, loc)
+
+      case Expression.InvokeStaticMethod(method, args, tpe, loc) =>
+        val as = args map (renameBoundVars(_, vars0, labels0))
+        Expression.InvokeStaticMethod(method, as, tpe, loc)
+
+      case Expression.GetField(field, exp, tpe, loc) =>
+        val e = renameBoundVars(exp, vars0, labels0)
+        Expression.GetField(field, e, tpe, loc)
+
+      case Expression.PutField(field, exp1, exp2, tpe, loc) =>
+        val e1 = renameBoundVars(exp1, vars0, labels0)
+        val e2 = renameBoundVars(exp2, vars0, labels0)
+        Expression.PutField(field, e1, e2, tpe, loc)
+
+      case Expression.GetStaticField(field, tpe, loc) => Expression.GetStaticField(field, tpe, loc)
+
+      case Expression.PutStaticField(field, exp, tpe, loc) =>
+        val e = renameBoundVars(exp, vars0, labels0)
+        Expression.PutStaticField(field, e, tpe, loc)
+
+      case Expression.NewChannel(exp, tpe, loc) =>
+        val e = renameBoundVars(exp, vars0, labels0)
+        Expression.NewChannel(e, tpe, loc)
+
+      case Expression.GetChannel(exp, tpe, loc) =>
+        val e = renameBoundVars(exp, vars0, labels0)
+        Expression.GetChannel(e, tpe, loc)
+
+      case Expression.PutChannel(exp1, exp2, tpe, loc) =>
+        val e1 = renameBoundVars(exp1, vars0, labels0)
+        val e2 = renameBoundVars(exp2, vars0, labels0)
+        Expression.PutChannel(e1, e2, tpe, loc)
+
+      case Expression.SelectChannel(rules, default, tpe, loc) =>
+        // TODO: is this right or do we need to handle it like Branch?
+        val rs = rules map {
+          case SelectChannelRule(sym, chan, exp) =>
+            val s = Symbol.freshVarSym(sym)
+            val vars1 = vars0 + (sym -> s)
+            val c = renameBoundVars(chan, vars0, labels0) // TODO: can sym occur in chan?
+            val e = renameBoundVars(exp, vars1, labels0)
+            SelectChannelRule(s, c, e)
+        }
+        val d = default.map(renameBoundVars(_, vars0, labels0))
+        Expression.SelectChannel(rs, d, tpe, loc)
+
+      case Expression.ProcessSpawn(exp, tpe, loc) =>
+        val e = renameBoundVars(exp, vars0, labels0)
+        Expression.ProcessSpawn(e, tpe, loc)
+
+      case Expression.ProcessPanic(msg, tpe, loc) => Expression.ProcessPanic(msg, tpe, loc)
+
+      case Expression.FixpointConstraintSet(cs0, tpe, loc) => Expression.FixpointConstraintSet(cs0, tpe, loc)
+
+      case Expression.FixpointCompose(exp1, exp2, tpe, loc) =>
+        val e1 = renameBoundVars(exp1, vars0, labels0)
+        val e2 = renameBoundVars(exp2, vars0, labels0)
+        Expression.FixpointCompose(e1, e2, tpe, loc)
+
+      case Expression.FixpointSolve(exp, stf, tpe, loc) =>
+        val e = renameBoundVars(exp, vars0, labels0)
+        Expression.FixpointSolve(e, stf, tpe, loc)
+
+      case Expression.FixpointProject(sym, exp, tpe, loc) =>  // TODO: rename?
```

Comment:
rename what?

Reply:
The sym variable. It's a different type of symbol than variable symbols or labels but I assume it still needs to be unique and can occur in exp?

Reply:
If you rebase onto master this is now called "name" and is not a symbol, just a string, so it there is no problem.